cmd/influxd/backup/backup.go

```diff
@@ -541,9 +541,15 @@ func (cmd *Command) download(req *snapshotter.Request, path string) error {
 			}
 
 			// Read snapshot from the connection
-			if n, err := io.Copy(f, conn); err != nil || n == 0 {
+			n, err := io.Copy(f, conn)
+			if err != nil {
 				return fmt.Errorf("copy backup to file: err=%v, n=%d", err, n)
 			}
+			if n == 0 {
+				// Unfortunately there is no out-of-band channel to actually return errors from the snapshot service, just
```

Comment:
In 2.x and 1.x Enterprise this API has a much improved wire format and will properly transmit errors. Not worth fixing in 1.x OSS, but we can at least point at the server side logs which will have the real error.